doc/man3/flux_kvs_lookup.adoc

```diff
@@ -53,6 +55,9 @@ the returned JSON is parsed according to variable arguments in Jansson
 `flux_kvs_lookup_get_raw()` is identical to `flux_kvs_lookup_get()` except
 the raw value is returned without decoding.
 
+`flux_kvs_lookup_get_dir()` is identical to `flux_kvs_lookup_get()` except
+a directory object is returned.
+
```

Comment:
I know you are considering removal of the `kvsdir` abstraction, but in the context of the addition of `flux_kvs_lookup_get_dir` it might be useful to consider addition of a refcount to kvsdir objects. This would avoid the necessary copies in the common case where we want the kvsdir to persist, but we're done with the future.

Reply:
We do have `flux_kvsdir_incref()` but since the object returned from the lookup is const (indicating that it still belongs to the future),  it seems a bit wrong to allow its reference count to be modified?  This is why I added `flux_kvsdir_copy()`, but I intended that to be short lived until the various uses where `flux_kvsdir_t` was used purposes other than "list of names in directory" were converted to something else TBD.

As we just discussed offline, possibly we need to come up with a new object that serves as a handle to the KVS, and contains things like a "current working directory", a snapshot reference, or a namespace handle, as well as the broker handle, and then all KVS functions would take one of those instead of the broker handle...  I'll open an issue on that idea.